api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java

```diff
@@ -339,4 +339,73 @@ public static Reference buildPatientReference(org.openmrs.Patient patient) {
 		patientReference.setId(patient.getUuid());
 		return patientReference;
 	}
+
+	/**
+	 * Compares patient objects with only current name and current address.
+	 * @param patient1
+	 * @param patient2
+	 * @return true if the patient are equalse, otherwise it returns false.
+	 */
+	public static boolean compareCurrentPatients(Object patient1, Object patient2) {
+		Patient p1 = (Patient) patient1;
+		Patient p2 = (Patient) patient2;
+
+		if (p1.getIdentifier().size() == p2.getIdentifier().size()) {
+			for (int i = 0; i < p1.getIdentifier().size(); i++) {
+				if (!p1.getIdentifier().get(i).equalsDeep(p2.getIdentifier().get(i))) {
+					return false;
+				}
+			}
+		}
+
+		if (p1.getName().size() > 0 && p2.getName().size() > 0) {
```

Comment:
Is it the work-around? I think that it should be comment in the code that will explain why you did this in this way